### PR TITLE
IMPROVE Worker DOCKER IMAGE: As Richard, I want the image for WORKER to be improved, so that it is smaller and contains fewer vulnerabilities.

### PR DESCRIPTION
**Pull request by Reinkaos**
** Acceptance Criteria **
- Dockerfile is brought in line with recent National Library and [DNZ security & efficiency improvements]

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,31 +1,54 @@
-FROM ruby:2.5.3
+FROM ruby:2.6.3-alpine AS builder
```
Comment (paul-mesnilgrente) [inline Dockerfile]: This is not matching the version below (2.6.3 and 2.6.5). You can use 2.6.5 :)

File: Dockerfile [hunk 2]
```diff
@@ -1,31 +1,54 @@
...
+    && bundle install --path=vendor/cache --without development:test:assets -j4 --retry 3 \
+    # Remove unneeded files (cached *.gem, *.o, *.c)
+    && rm -rf $GEM_HOME/cache/*.gem \
+    && find $GEM_HOME/gems/ -name "*.c" -delete \
+    && find $GEM_HOME/gems/ -name "*.o" -delete
 
-# Use libxml2, libxslt a packages from alpine for building nokogiri
-RUN bundle config build.nokogiri --use-system-libraries
+# Change TimeZone
+RUN apk add --update tzdata
```
Comment (paul-mesnilgrente) [inline Dockerfile]: `tzdata` is already in the list of the installed packages, there's no need to re-install it. Also, setting it here in the build step will be ineffective as the "real" docker image is built below.

File: Dockerfile [hunk 3]
```diff
@@ -1,31 +1,54 @@
...
-WORKDIR /var/worker
-COPY Gemfile Gemfile
-COPY Gemfile.lock Gemfile.lock
-RUN bundle install --binstubs --without development test --path vendor/cache
-COPY . /var/worker
+ARG RAILS_ENV="production"
+ARG SECRET_KEY_BASE
 
+ENV RAILS_ENV=$RAILS_ENV
+ENV SECRET_KEY_BASE=$SECRET_KEY_BASE
```
Comment (paul-mesnilgrente) [inline Dockerfile]: That's useful in other apps to have the correct context in the asset pipeline but as there's no assets in the worker, that can be removed :)

File: Dockerfile (hunk 2 above)
Comment (Reinkaos) [reply Dockerfile]: Nice! Yeah I think I just copied that one across, but totally not needed.

File: Dockerfile (hunk 3 above)
Comment (Reinkaos) [reply Dockerfile]: Yeah :)

**Pull request merged**